cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -222,51 +222,55 @@ where
     for transaction in block.transactions.into_iter() {
         let tx_hash = transaction.hash;
 
-        let receipt = if needs_receipt {
-            let receipt = connector
-                .receipt_by_hash(tx_hash)
-                .compat()
-                .await?
-                .ok_or_else(|| {
-                    anyhow::anyhow!(
-                        "Could not get transaction receipt for transaction {:x}",
-                        tx_hash
-                    )
-                })?;
-
-            Some(receipt)
-        } else {
-            None
-        };
-
-        let result = pattern.matches(&transaction, receipt.as_ref());
+        let receipt = connector
+            .receipt_by_hash(tx_hash)
+            .compat()
+            .await?
+            .ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Could not get transaction receipt for transaction {:x}",
+                    tx_hash
+                )
+            })?;
 
         tracing::debug!(
-            "matching {:?} against transaction {:x} yielded {}",
-            pattern,
+            "checking transaction {:x} status: {:?}",
             tx_hash,
-            result
+            receipt.status
         );
 
-        if result {
-            let receipt = match receipt {
-                Some(receipt) => receipt,
-                None => connector
-                    .receipt_by_hash(tx_hash)
-                    .compat()
-                    .await?
-                    .ok_or_else(|| {
-                        anyhow::anyhow!(
-                            "Could not get transaction receipt for transaction {:x}",
-                            tx_hash
-                        )
-                    })?,
-            };
-
-            return Ok(Some(TransactionAndReceipt {
-                transaction,
-                receipt,
-            }));
+        if receipt.status == Some(1.into()) {
```

Comment:
Can we have a const with descriptive name please instead of `1.into()`, perhaps `const ACCEPTED = 1.into()` (type missing :)